### ISSUE TITLE
Commented Annotations with 4 Columns are Still Applied

### ISSUE THREAD

**Issue by emmahodcroft** [nextstrain/ncov-ingest#95]
**Current Behavior**  
As spotted by @eharkins - currently it seems rows that are commented but have 4 columns are applied, rather than ignored.

In [this commit](https://github.com/nextstrain/ncov-ingest/commit/69cde21eb4a2687a0e3e4bc31b4ca0b7970d15d0), lines 698-707, Eli included some commented information for reference but that shouldn't be processed. 

Example:
```
# USA/OK-ADDL-03/2020	EPI_ISL_535362	location	City Suburb
# USA/OK-ADDL-04/2020	EPI_ISL_535363	location	City Suburb
# USA/PR-B35B/2020	EPI_ISL_534693	location	Ausilio Mutuo # Hospital in San Juan
# USA/PR-B3SB/2020	EPI_ISL_534694	location	Ausilio Mutuo # Hospital in San Juan
```

However, after pushing this update, we got notified that these locations had been applied, despite being commented:
![image](https://user-images.githubusercontent.com/14290674/93356747-dd70b280-f83f-11ea-8241-a431a748976f.png)

**Expected behavior**  
Any line that starts with `#` should be fully ignored, even if it has the right number of columns.

**Possible solution**  
This seems to stem from [this section](https://github.com/nextstrain/ncov-ingest/blob/master/bin/transform-gisaid#L97) of the code in `gisaid_transform` where we check if a line has 4 columns -- only if it _does not_ then do we see if it is commented. If it has 4 columns, it seems to go ahead and try to process it, even if it starts with `#`. 

We should change this so that it first checks to see if the first character (minus whitespace) is `#` and ignore the whole line if so (regardless of number of columns).

**Temporary solution**
Ensure that commented columns we don't want to be applied do *not* have 4 columns.

**Comment by emmahodcroft** [comment]
I will try to work on this later if I have a minute - just trying to do something else (non-COVID-related!) at the moment. If not, I imagine @kairstenfay or @ttung would be able to check into this and (probably) fix pretty quickly!

**Comment by emmahodcroft** [comment]
Finally this did raise one more question for me - we have other lines (older) that are commented but have 4 columns. However, they reference `key`s that don't exist - take for example this row:
```
# USA/FL-9F97/2020	EPI_ISL_513265	additional_host_info: Tranplant Patient	additional_location_info: Jackson Memorial Hospital
```

Shouldn't we be getting an error showing up saying that it couldn't apply a `key` of 'additional_host_info Tranplant Patient'?

**Comment by ttung** [comment]
> Shouldn't we be getting an error showing up saying that it couldn't apply a key of 'additional_host_info Tranplant Patient'?

No, I didn't end up adding that code, though we probably should.

Also, does that have 4 tab-separated columns?  It looks like spaces.

**Comment by ttung** [comment]
See #96 for the fix for this particular issue.

**Comment by emmahodcroft** [comment]
Thanks @ttung! I can make a separate issue for the warnings thing - sorry, I had thought this was in there! 

>Also, does that have 4 tab-separated columns? It looks like spaces.

Yea, hard to copy to Github and make this look convincing. However, can confirm they are tabs, via my fancy text editor color-column plugin :)
![image](https://user-images.githubusercontent.com/14290674/93360385-e5325600-f843-11ea-850f-184fb3c240f5.png)

**Comment by emmahodcroft** [comment]
Fixed by #96 !

**Comment by emmahodcroft** [comment]
I've filed a new issue about the warnings in #97. I don't think there's urgency on this. I'll try to do a quick test today to estimate whether this is a problem at all in our current file and if so, the size of it
